roles/pulp_common/tasks/main.yml

```diff
@@ -32,6 +32,12 @@
   loop_control:
     loop_var: __pulp_common_req_var
 
+- name: Check required roles if FIPS detected
+  assert:
+    that:
+      - "'pulp_devel' in role_names"
+  when: ansible_fips
```

Comment:
Now i'm confused, as you don't check ansible_fips on the other task anymore.

Reply:
Because that's what @bmbouter asked for. You wanted an early detection kill switch and he wanted the patch to always occur when devel role is specified. Obviously it can't go into the devel role or else the installer would fail under FIPS.